h2o-docs/src/product/data-science/glm.rst

```diff
@@ -1740,7 +1741,47 @@ Regress :math:`z_{i}` on the predictors :math:`x_{i}` using the weights :math:`w
 
 This process is repeated until the estimates :math:`\hat{\beta}` change by less than the specified amount.
 
-**Cost of computation**
+**Tweedie Likelihood Calculation**
+
+You only calculate Tweedie likelihood when you estimate the Tweedie variance power or the Tweedie variance power with dispersion. The calculation in this section is used to estimate the full log likelihood. When you fix the Tweedie variance power, you will use a simpler formula (unless you are estimating dispersion). When fixing the Tweedie variance power for dispersion estimation, you use the Series method.
+
+Depending on :math:`p`, :math:`y`, and :math:`\phi`, different methods are used for this log likelihood estimation. To start, let:
+
+.. math::
+   
+   \xi = \frac{\phi}{y^{2-p}}
+
+If :math:`p=2`, then it will use the log likelihood of the Gamma distribution:
+
+.. math::
+
+   \log (p) = \begin{cases} - \infty & y=0 \\ \frac{1}{\phi} \log (\frac{1}{\phi \mu}) - \log \text{Gamma} \frac{1}{\phi} + \log (y)(\frac{1}{\phi} -1) + (-\frac{1}{\phi \mu} y) & y>0 \\\end{cases}
+
+If :math:`p=3`, then it will use the inverse Gaussian distribution:
+
+.. math::
+   
+   \log (p) = \begin{cases} - \infty & y=0 \\ \frac{1}{2} \Big(-log (\phi \mu) \log (2 \pi) -3 \log \big( \frac{y}{\mu} - \frac{(\frac{y}{\mu} -1)^2}{\phi \mu \frac{y}{\mu}} \Big) - \log (\mu) & y>0 \\\end{cases}
+
+If :math:`p<2` and :math:`\xi \leq 0.01`, then it will use the Fourier inversion method.
+
+If :math:`p>2` and :math:`\xi \geq 1`, then it will also use the Fourier inversion method.
+
+Everything else will use the Series method. However, if the Series method fails (output of ``NaN``), then it will try the Fourier inversion method instead.
+
+If both the Series method and Fourier inversion method fail, or if the Fourier inversion method was chosen based on the :math:`\xi` criterium and it failed, it will then estimate the log likelihood using the Saddlepoint approximation.
+
+Here is the general usage for Tweedie variance power maximum likelihood:
+
+- ``fix_tweedie_variance_power = False`` and ``fix_dispersion_parameter = True`` as it will optimize the Tweedie variance power
+- ``fix_tweedie_variance_power = False`` and ``fix_dispersion_parameter = False`` as it will optimize both Tweedie variance power and the dispersion parameter
+
+*Optimization Procedure*
+
+When estimating just the Tweedie variance power, it use the golden section search. Once a small region is found, then it switches to Newton's method. If Newton's method fails (i.e. steps out of the bounds found by the golden section search), it uses the golden section search until convergence. When you optimize both Tweedie variance power and dispersion, it uses the Nelder-Mead method with constraints so that Tweedie variance power :math:`p>1+10^{-10}` and dispersion :math:`\phi >10^{-10}`.
```

Comment:
```suggestion
When estimating just the Tweedie variance power, it uses the golden section search. Once a small region is found, then it switches to Newton's method. If Newton's method fails (i.e. steps out of the bounds found by the golden section search), it uses the golden section search until convergence. When you optimize both Tweedie variance power and dispersion, it uses the Nelder-Mead method with constraints so that Tweedie variance power :math:`p>1+10^{-10}` and dispersion :math:`\phi >10^{-10}`. If the Nelder-Mead seems to be stuck in local optimum, you might want to try increasing the `dispersion_learning_rate`.
```